activerecord/lib/active_record/tasks/sqlite_database_tasks.rb

```diff
@@ -30,8 +30,8 @@ def drop
       end
 
       def purge
-        drop
         connection.disconnect!
```

Comment:
Calling `connection` recreates the database which lead to errors further down the codepath (`raise DatabaseAlreadyExists if File.exist?(db_config.database)` actually raising when `ensure { create }` (L37) is called.